### ISSUE TITLE
Lint/Syntax run wrong result frequently. 

### ISSUE THREAD

**Issue by zw963** [rubocop/rubocop#10509]
Sorry for no much clue to describe this issue, basically, when Enable rubocop with lsp-mode (use solargraph), it work quite well, do format very quickly too, but, it often give me wrong feedback for syntax error. 

e.g. (see following screenshot)

![image](https://user-images.githubusercontent.com/549126/162280188-f27dc116-aa70-4b03-87f3-b7dbab074b4a.png)

in this case, there is no syntax error, but lsp+rubocop give me some wrong result.

If i restart emacs (or emacs daemon), and open this file again, it get correct result now.

so, what should i do for next this issue happen again for get some useful info? 

thank you.

**Comment by zw963** [comment]
Following is some config:

```el
(require 'lsp-solargraph)

(defun zw/lsp-ruby-common-hooks ()
  (rvm-activate-corresponding-ruby)
  (add-hook 'before-save-hook #'lsp-format-buffer t t)
  (lsp-deferred)
  )

(add-hook 'ruby-mode-hook 'zw/lsp-ruby-common-hooks)
(add-hook 'enh-ruby-mode-hook 'zw/lsp-ruby-common-hooks)

(require 'rubocop)

(setq rubocop-check-command "rubocop-daemon-wrapper --format emacs")
(setq rubocop-autocorrect-command "rubocop-daemon-wrapper -a --format emacs")
(setq rubocop-format-command "rubocop-daemon-wrapper -x --format emacs")

(add-hook 'ruby-mode-hook #'rubocop-mode)
```

**Comment by zw963** [comment]
this issue happen a gain, i never see those error message in lsp io buffer when i follow following step.

```
Set lsp-log-io to t to inspect communication between client and the server. 
Use lsp-workspace-show-log to switch to the corresponding log buffer.
```

And, i no see any Offenses relative to this issue.

But, the lsp keeping give me error (please see screenshot)

![image](https://user-images.githubusercontent.com/549126/162690060-7d199ba8-a933-496f-aa27-1eed37eb7804.png)

![image](https://user-images.githubusercontent.com/549126/162690104-69892ead-ff02-4538-ab0b-4afc195ccc89.png)

Because i consider this issue come from following cops

```yml
Lint/Syntax:
  Description: 'Checks for syntax errors.'
  Enabled: true
  VersionAdded: '0.9'
```

And, it not possible to disable it in ~/.rubocop.yml or project's rubocop.yml

```
Error: configuration for Syntax cop found in .rubocop.yml
It's not possible to disable this cop.
```

So, could you please give me some clue about how to investigate this?

thank you.
